core/app/models/spree/order.rb

```diff
@@ -190,6 +190,11 @@ def self.not_canceled
       where.not(state: 'canceled')
     end
 
+    # => Get all nonzero (reference to :nonzero scope of Spree::Adjustment) adjustments for an instance of Spree::Order and its LineItem and Shipment associations
+    def absolute_adjustments
+      all_adjustments.nonzero.or(Spree::Adjustment.where(adjustable: self).nonzero)
```

Comment:
I'm having trouble understanding in what situation is the `order_id` not set to the current order on an order-level adjustment. `all_adjustments` uses the `order_id` foreign key, so it should return all the adjustments associated with the order, regardless of whether the adjustable is the order itself or a shipment/line item.

Reply:
I'm also concerned about naming here. Here are my concerns:

1. I think this method should probably be called `nonzero_adjustments`. An "absolute" value of x is ["the non-negative value of x without regard to its sign"](https://en.wikipedia.org/wiki/Absolute_value) which isn't really a good description of this. These are just "non-zero" adjustments. We already use the term `nonzero` to describe them elsewhere so I would need to see a good argument for introducing a second name of `nonzero` to the codebase.
2. To be consistent with the naming of the associations (`adjustments` are the order-level adjustments, `all_adjustments` are all the adjustments associated with the order in any way) this method should probably be called `all_nonzero_adjustments`.
3. If we were to call this method `all_nonzero_adjusments`, there's not much value in this method. To me at least, `order.all_nonzero_adjustments` isn't any easier to write or remember than `order.all_adjustments.nonzero`.

Reply:
> I'm having trouble understanding in what situation is the `order_id` not set to the current order on an order-level adjustment. `all_adjustments` uses the `order_id` foreign key, so it should return all the adjustments associated with the order, regardless of whether the adjustable is the order itself or a shipment/line item.

That's not wrong. The call to the method could be replaced by `order.all_adjustments.nonzero.any?`. However, when I wrote this I was thinking of passing the same arguments to the query that Solidus was already doing and that's why there's an `.or` uniting already existent scopes.

I'll change locally and see if it yields the same results.